### ISSUE TITLE
Save only the model with the best validation loss and from the latest epoch

### ISSUE THREAD

**Issue by HansBambel** [HansBambel/SmaAt-UNet#44]
To save disk space and make it easier for users after training to identify the best model it would be good to only save the model with the lowest validation loss and also from the last epoch to resume training if one desires.

This can be helpful: https://lightning.ai/docs/pytorch/stable/common/checkpointing_intermediate.html#which

**Comment by rutkovskii** [comment]
On it. Once I verify that it works correctly, I will submit pull request.

**Comment by HansBambel** [comment]
Added with #45 thanks to @rutkovskii
